File: web/src/main/java/by/sorface/passport/web/dao/nosql/redis/RedisOAuth2AuthorizationCompleteRepository.java
package by.sorface.passport.web.dao.nosql.redis;

import by.sorface.passport.web.dao.nosql.redis.models.RedisOAuth2AuthorizationComplete;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.QueryByExampleExecutor;

import java.util.Optional;

public interface RedisOAuth2AuthorizationCompleteRepository extends CrudRepository<RedisOAuth2AuthorizationComplete, String>, QueryByExampleExecutor<RedisOAuth2AuthorizationComplete> {

    Optional<RedisOAuth2AuthorizationComplete> findFirstByAccessToken(final String accessToken);

    Optional<RedisOAuth2AuthorizationComplete> findFirstByRefreshToken(final String refreshToken);

    Optional<RedisOAuth2AuthorizationComplete> findFirstByOidcToken(final String oidcToken);

}

File: web/src/main/kotlin/by/sorface/passport/web/dao/nosql/redis/RedisOAuth2AuthorizationCompleteRepository.kt
package by.sorface.passport.web.dao.nosql.redis

import by.sorface.passport.web.dao.nosql.redis.models.RedisOAuth2AuthorizationComplete
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.QueryByExampleExecutor
import java.util.*

interface RedisOAuth2AuthorizationCompleteRepository : CrudRepository<RedisOAuth2AuthorizationComplete, String>,
    QueryByExampleExecutor<RedisOAuth2AuthorizationComplete> {

    fun findFirstByAccessToken(accessToken: String): RedisOAuth2AuthorizationComplete?

    fun findFirstByRefreshToken(refreshToken: String): RedisOAuth2AuthorizationComplete?

    fun findFirstByOidcToken(oidcToken: String): RedisOAuth2AuthorizationComplete?

}
